cli/command/service_test.go

```diff
@@ -0,0 +1,512 @@
+	for name, c := range contexts {
+		t.Run(name, func(t *testing.T) {
+			if err := serviceCommand.logs(c); err == nil {
+				t.Fatal("%s: error was nil!", name)
+			}
+		})
+	}
+}
+
+func TestReadService(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	base.Resolver.EXPECT().
+		Resolve("service", "svc_name").
+		Return([]string{"svc_id"}, nil)
+
+	base.Client.EXPECT().
+		ReadService("svc_id").
+		Return(&models.Service{}, nil)
+
+	args := Args{"svc_name"}
+	c := NewContext(t, args, nil)
+
+	serviceCommand := NewServiceCommand(base.Command())
+	if err := serviceCommand.read(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestReadService_userInputError(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg": NewContext(t, nil, nil),
+	}
+
+	serviceCommand := NewServiceCommand(base.Command())
+	for name, c := range contexts {
+		t.Run(name, func(t *testing.T) {
+			if err := serviceCommand.read(c); err == nil {
+				t.Fatal("%s: error was nil!", name)
+			}
+		})
+	}
+}
+
+func TestScaleService(t *testing.T) {
+	defer SetTimeMultiplier(0)()
+
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	base.Resolver.EXPECT().
+		Resolve("service", "svc_name").
+		Return([]string{"svc_id"}, nil)
+
+	scale := 2
+	req := models.UpdateServiceRequest{
+		ServiceID: "svc_id",
+		Scale:     &scale,
+	}
+
+	base.Client.EXPECT().
+		UpdateService(req).
+		Return("job_id", nil)
+
+	job := &models.Job{
+		Status: "Completed",
+		Result: "svc_id",
+	}
+
+	base.Client.EXPECT().
+		ReadJob("job_id").
+		Return(job, nil)
+
+	deployments := []models.Deployment{
+		{
+			DesiredCount: 1,
+			RunningCount: 1,
+		},
+	}
+
+	service := &models.Service{
+		Deployments:  deployments,
+		DesiredCount: 1,
+		RunningCount: 1,
+	}
+
+	base.Client.EXPECT().
+		ReadService("svc_id").
+		Times(4).
+		Return(service, nil)
+
+	args := Args{"svc_name", "2"}
+	c := NewContext(t, args, nil)
+	serviceCommand := NewServiceCommand(base.Command())
+	if err := serviceCommand.scale(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestScaleService_noWait(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	base.Resolver.EXPECT().
+		Resolve("service", "svc_name").
+		Return([]string{"svc_id"}, nil)
+
+	scale := 2
+	req := models.UpdateServiceRequest{
+		ServiceID: "svc_id",
+		Scale:     &scale,
+	}
+
+	base.Client.EXPECT().
+		UpdateService(req).
+		Return("job_id", nil)
+
+	args := Args{"svc_name", "2"}
+	c := NewContext(t, args, nil, SetNoWait(true))
+
+	serviceCommand := NewServiceCommand(base.Command())
+	if err := serviceCommand.scale(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestScaleService_userInputError(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg":  NewContext(t, nil, nil),
+		"Missing COUNT arg": NewContext(t, Args{"svc_name"}, nil),
```

Comment:
should have a case for non-integer COUNT arg

Reply:
✅